### PR TITLE
document ansible_group_priority

### LINKED ISSUE #28769: add docs re ansible_group_priority
##### ISSUE TYPE
 - Documentation Report

##### COMPONENT NAME
ansible_group_priority

##### ANSIBLE VERSION
```
ansible-playbook 2.4.0 (devel 81caa837a4) last updated 2017/08/29 15:00:19 (GMT +100)
```

##### CONFIGURATION
NA

##### OS / ENVIRONMENT
NA

##### SUMMARY
ansible_group_priority is mentioned in the changelog for 2.4, but not documented anywhere.

##### STEPS TO REPRODUCE
NA

##### EXPECTED RESULTS
Should be able to search and find a description of what it is with an example of how to use it.

##### ACTUAL RESULTS
No search results returned.

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY

documenting new feature
fixes #28769

##### ISSUE TYPE

 - Docs Pull Request

##### COMPONENT NAME

inventory
##### ANSIBLE VERSION

```
2.4
```

### CODE REVIEW

**Review by dharmabumstead — APPROVED** [review]
Approved with copy edits. Thanks @bcoca!

File: docs/docsite/rst/intro_inventory.rst [hunk 1]
```diff
@@ -222,12 +222,40 @@ directory will override variables set in the inventory directory.
...
+
+In this example, if both groups have the same priority, the result would normally have been ``testvar == b``, but since we are giving the ``a_group`` a higher priority the result will be ``testvar == a``.
+
 .. _behavioral_parameters:
 
 List of Behavioral Inventory Parameters
 +++++++++++++++++++++++++++++++++++++++
 
-As alluded to above, setting the following variables controls how ansible interacts with remote hosts.
+As described above, setting the following variables control how Ansible interacts with remote hosts.
```
Comment (cognifloyd) [inline docs/docsite/rst/intro_inventory.rst:258]: `setting [these] controls` = `setting the following variables controls`
or
`the following variables control` (drop "setting")

File: docs/docsite/rst/playbooks_variables.rst [hunk 2]
```diff
@@ -867,6 +867,8 @@ Basically, anything that goes into "role defaults" (the defaults folder inside t
           If multiple groups have the same variable, the last one loaded wins.
           If you define a variable twice in a play's vars: section, the 2nd one wins.
 .. note:: the previous describes the default config `hash_behavior=replace`, switch to 'merge' to only partially overwrite.
+.. note:: Group loading follows parent/child relationships. Groups of the same 'patent/child' level are then merged following alphabetical order.
```
Comment (cognifloyd) [inline docs/docsite/rst/playbooks_variables.rst:870]: s/patent/parent/

**Comment by kaistian** [comment]
This PR is missing from stable 2.4 branch and therefore it's missing from docs.ansible.com, could it be merge into stable please?

**Comment by termlen0** [comment]
Has this been tested by setting group_variable_priority inside a `group_vars` file as well? I might be doing something wrong, but the following does not work:
```(venv2) ~/P/m/ansible_2.4_updates ❯❯❯ cat group_vars/group1.yaml
test_var: 20
ansible_group_priority: 1

```
Whereas, setting this within the inventory file as:

```
group1:
  hosts:
    device1
  vars:
    test_var: higher
    ansible_group_priority: 10

group2:
  hosts:
    device1
  vars:
    test_var: lower
```
Works as expected.

**Comment by gilesw** [comment]
@termlen0 @bcoca I've found the same thing too. I was trying to set the variable using in an ini format inventory.

**Comment by hanej** [comment]
The documentation doesn't show you how to use this feature.  Where does this go?  Can I use this in group_vars or in the inventory file?  Are there examples that show how this works?

**Comment by frntn** [comment]
@hanej I have found this https://github.com/ansible/ansible/blob/76ff3e9efc0a74799bc2962508fdfcc8c2a920b6/docs/docsite/rst/user_guide/intro_inventory.rst#how-variables-are-merged

But not sure how to make use it in a `ini` format inventory...

**Pull request merged**